### PR TITLE
Email template changes

### PR DESCRIPTION
**Pull request by nithyal1**

### CODE REVIEW

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AzureServicePrinicipalAccountsService.java [hunk 1]
```diff
@@ -3406,4 +3407,236 @@ private boolean rotateAzureServicePrincipalSecret(String token, String servicePr
 			return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("{\"errors\":[\"Access denied: No permission to remove AWS Role from this Azure service account\"]}");
 		}
 	}
+	
+	/**
+     * Remove Group from IAM Service Account
+     *
+     * @param token
+     * @param iamServiceAccountGroup
```
Comment (sarathmohan4) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AzureServicePrinicipalAccountsService.java:3415]: iam?

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AzureServicePrinicipalAccountsService.java [hunk 2]
```diff
@@ -3406,4 +3407,236 @@ private boolean rotateAzureServicePrincipalSecret(String token, String servicePr
...
+        else {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("{\"errors\":[\"Access denied: No permission to remove groups from this Azure service principal\"]}");
+        }
+
+    }
+    
+    /**
+	 * Method to update policies for remove group from IAM service account.
+	 * @param token
+	 * @param iamServiceAccountGroup
```
Comment (sarathmohan4) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AzureServicePrinicipalAccountsService.java:3488]: iam

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AzureServicePrinicipalAccountsService.java [hunk 3]
```diff
@@ -3406,4 +3407,236 @@ private boolean rotateAzureServicePrincipalSecret(String token, String servicePr
...
+			ldapConfigresponse = oidcUtil.updateGroupPolicies(token, azureServiceAccountGroup.getGroupname(), policies, currentpolicies,
+					oidcGroup != null ? oidcGroup.getId() : null);
+			oidcUtil.renewUserToken(userDetails.getClientToken());
+		}
+		if(ldapConfigresponse.getHttpstatus().equals(HttpStatus.NO_CONTENT) || ldapConfigresponse.getHttpstatus().equals(HttpStatus.OK)){
+			return updateMetadataForRemoveGroupFromAzureSvcAcc(token, azureServiceAccountGroup, userDetails, oidcGroup,
+					currentpolicies, currentpoliciesString);
+		}
+		else {
+		    return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("{\"errors\":[\"Failed to remove the group from the IAM Service Account\"]}");
```
Comment (sarathmohan4) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AzureServicePrinicipalAccountsService.java:3556]: IAM,

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AzureServicePrinicipalAccountsService.java [hunk 4]
```diff
@@ -3406,4 +3407,236 @@ private boolean rotateAzureServicePrincipalSecret(String token, String servicePr
...
+		}
+	}
+	
+	/**
+	 * Method to update metadata and revert group permission if metadata update failed.
+	 *
+	 * @param token
+	 * @param iamServiceAccountGroup
```
Comment (sarathmohan4) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AzureServicePrinicipalAccountsService.java:3564]: IAM

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AzureServicePrinicipalAccountsService.java [hunk 5]
```diff
@@ -3406,4 +3407,236 @@ private boolean rotateAzureServicePrincipalSecret(String token, String servicePr
...
+			return revertGroupPermissionForAzureSvcAcc(token, azureServiceAccountGroup, userDetails, oidcGroup,
+					currentpolicies, currentpoliciesString, metadataResponse);
+		}
+	}
+	
+	/**
+	 * Method to revert group permission for IAM service account
+	 *
+	 * @param token
+	 * @param iamServiceAccountGroup
```
Comment (sarathmohan4) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AzureServicePrinicipalAccountsService.java:3600]: IAM

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/AzureServicePrinicipalAccountsController.java [hunk 6]
```diff
@@ -269,5 +269,18 @@
 	@ApiOperation(value = "${AzureServicePrinicipalAccountsController.rotateSecret.value}", notes = "${AzureServicePrinicipalAccountsController.rotateSecret.notes}")
 	public ResponseEntity<String> rotateSecret(HttpServletRequest request, @RequestHeader(value="vault-token") String token, @RequestBody @Valid AzureServicePrinicipalRotateRequest azureServicePrinicipalRotateRequest){
 		return azureServicePrinicipalAccountsService.rotateSecret(token, azureServicePrinicipalRotateRequest);
+	}	
+	/**
+	 * Removes permission for a group from the IAM service account
+	 * @param request
+	 * @param token
+	 * @param iamServiceAccountGroup
```
Comment (sarathmohan4) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/AzureServicePrinicipalAccountsController.java:277]: IAM

**Review by sarathmohan4 — APPROVED** [review]

**Review by alexrajan3 — APPROVED** [review]

**Pull request merged**